docs/GettingStarted.md

```diff
@@ -268,9 +273,22 @@ Install and configure the Datadog Agent to receive traces from your now instrume
     end
     ```
 
-4. (Optional) Add or activate additional instrumentation by doing either of the following:
-    - Activate Datadog integration instrumentation (see [Integration instrumentation](#integration-instrumentation))
-    - Add Datadog manual instrumentation around your code (see [Manual instrumentation](#manual-instrumentation))
+     Using this block you can:
+
+      - [Add additional configuration settings](#additional-configuration)
+      - [Activate or reconfigure instrumentation](#integration-instrumentation)
+
+
+### Connect your application to the Datadog Agent
+
+`ddtrace` will submit traces to the agent via UDS by default (if a socket is discovered.) Otherwise, it will fallback and submit traces via HTTP to `127.0.0.1:8126`.
```

Comment:
```suggestion
`ddtrace` will submit traces to the agent via UDS by default (if a socket is discovered). Otherwise, it will fallback and submit traces via HTTP to `127.0.0.1:8126`.
```
Can you confirm that this case is only when the trace agent target is not set? If it is the case it can be mentioned there

Reply:
Not sure what you mean by `only when the trace agent target is not set`?

Logic is simple: it looks for socket file. If it finds it at the default location it uses it. Otherwise it uses configured options. If no options are configured it uses the defaults for those, which is `127.0.0.1:8126`.

Reply:
So IIUC it will use an autodetected socket over any user-set configuration?

Reply:
This is in agent settings resolver:

```ruby
def adapter
  # If no agent settings have been provided, we try to connect using a local unix socket.
  # We only do so if the socket is present when `ddtrace` runs.
  if should_use_uds_fallback?
    Transport::Ext::UnixSocket::ADAPTER
  else
    Transport::Ext::HTTP::ADAPTER
  end
end

def should_use_uds_fallback?
  uds_fallback != nil
end

# We only use the default unix socket if it is already present.
# This is by design, as we still want to use the default host:port if no unix socket is present.
def uds_fallback
  return @uds_fallback if defined?(@uds_fallback)

  @uds_fallback =
    if configured_hostname.nil? &&
        configured_port.nil? &&
        deprecated_for_removal_transport_configuration_proc.nil? &&
        File.exist?(Transport::Ext::UnixSocket::DEFAULT_PATH)

      Transport::Ext::UnixSocket::DEFAULT_PATH
    end
end

DEFAULT_PATH = '/var/run/datadog/apm.socket'.freeze
```

So looks like if a hostname/port is explicitly configured, then it will use that over UDS. But if nothing is configured, and it discovers a socket, it will use that.

Reply:
I changed the wording on this section to make it clear it uses configured settings first, otherwise falls back to UDS, and then HTTP.